src/fromager/wheels.py

```diff
@@ -268,11 +266,14 @@ def build_wheel(
         )
         extra_environ["DIST_EXTRA_CONFIG"] = str(dist_extra_cfg)
 
-    # add package and variant env vars last
+    # add package and variant env vars
     template_env = os.environ.copy()
     template_env.update(extra_environ)
     extra_environ.update(pbi.get_extra_environ(template_env=template_env))
 
+    # activate virtual env
+    build_env.add_venv_environ(extra_environ)
+
```

Comment:
This section is confusing. It seems to be updating multiple dicts with similar information, iteratively? 

I think I understand that `pbi.get_extra_environ()` needs the template environment in order to evaluate the template for some of the values. Are we sure that will never need to include `$PATH`, `$VIRTUAL_ENV`, or any of the values defined in `extra_environ` above? In particular, I could see `MAX_JOBS` being useful if the way to specify that for a build tool is using a different variable name.

If `add_venv_environ()` just returned those settings, then could we put them into `template_env` before calling `get_extra_environ()`? Or would we have to worry about the ordering?

Reply:
You are right, the logic was confusing. I wasn't happy with the previous implementation, either. It's inconsistent, too. For example the parallel job settings were only applied in the build wheel path, not in other paths. The package env templates did not have access to correct `VIRTUAL_ENV` either.

Now `get_extra_environ()` applies the parallel job settings and takes an optional `build_env` arg to apply virtual env settings from the build environment.